augur/reconstruct_sequences.py

```diff
@@ -73,7 +74,7 @@ def run(args):
     if(is_vcf):
         node_data["nodes"][root_node]['aa_sequences'] = {}
         with open_file(args.vcf_aa_reference) as handle:
-            for record in SeqIO.parse(handle, "fasta"):
+            for record in read_sequences(handle):
```

Comment:
I might prefer "explicit is better than implicit" here, better to be explicit than rely on implicit default args?

Reply:
Considering this comment and https://github.com/nextstrain/augur/pull/1731#discussion_r1922743349, my sense is a proper solution would abstract away the Biopython format for callers of `augur.io`. This could be done via the `read_fasta_sequences`+`read_genbank_sequences` split that you've mentioned, but that adds unnecessary verbosity in amount of functions after including the singular counterparts `read_fasta_sequence`+`read_genbank_sequence`.

We could just keep using `format="fasta"` but change the way that it is used, away from simply passed to Biopython to explicitly either `"genbank"` or `"fasta"` which conditionally translates to `"fasta-pearson"` in `read_sequence(s)`. I'll do that in the other PR.

Reply:
Addressed with d315bdb1ec7920351e891fa92ede26f16d16d925